libsolidity/ast/Types.cpp

```diff
@@ -1264,6 +1264,42 @@ FixedPointType const* RationalNumberType::fixedPointType() const
 	);
 }
 
+pair<RationalNumberType const*, RationalNumberType const*> RationalNumberType::fractionalDecomposition() const
+{
+	rational const maxUint = rational((bigint(1) << 256) - 1);
+	rational const minInt = -rational(bigint(1) << 255);
+
+	bool negative = (m_value < 0);
+	rational const maxMantissa = (negative ? -minInt : maxUint);
+
+	rational negatedExponent = 0;
+	rational unsignedMantissa = abs(m_value);
+
+	if (unsignedMantissa > maxMantissa)
+		return {nullptr, TypeProvider::rationalNumber(0)};
+
+	while (unsignedMantissa.denominator() != 1)
+	{
+		unsignedMantissa *= 10;
+		++negatedExponent;
+
+		// NOTE: Technically RationalNumberType::isValidLiteral() should not allow an exponent so
+		// large we cannot store it in 4096 bits. For some reason it does not validate numbers not
+		// in scientific notation though. See https://github.com/ethereum/solidity/issues/14100
```

Comment:
I don't see how any of this is relevant here.

Reply:
The fact that we're even having this discussion and you need to explain this to me means that at the very least this is not properly documented/commented in the source and is a time waster for anyone who encounters this weird behavior.

But anyway, I don't want to continue this discussion so I removed the comment.